### ISSUE TITLE
Improve error message of pre-flight check when connection to Github can not be established

### ISSUE THREAD

**Issue by gbraad** [minishift/minishift#2102]
```
$ echo | sudo tee /etc/resolv.conf
$ ping github.com
ping: github.com: Name or service not known
$ minishift delete --force; minishift start --iso-url minikube
-- Starting profile 'minishift'
-- Checking if requested OpenShift version 'v3.7.1' is valid ... FAIL
   v3.7.1 is not a valid OpenShift version
```

I would expect a different error message to handle this error case.

**Comment by arilivigni** [comment]
@gbraad This is currently blocking us on new installs.  Is there a way we can specify a location of a cached version to get around this or are you planning a fix for this very soon?

**Comment by LalatenduMohanty** [comment]
@arilivigni You can use the work around `$ minishift config set skip-check-openshift-release true`

Refer https://github.com/minishift/minishift/issues/2083

**Comment by arilivigni** [comment]
@LalatenduMohanty I still hit this running on an Openstack VM with that setting.

/home/cloud-user/minishift/minishift start --profile minishift --disk-size 40gb --memory 6400mb --iso-url file:////home/cloud-user/minishift/minishift.iso --show-libmachine-logs -v5
-- minishift version: v1.14.0+1ec5877
-- Starting profile 'minishift'
-- Checking if requested OpenShift version 'v3.7.1' is valid ... SKIP
-- Checking if requested OpenShift version 'v3.7.1' is supported ... SKIP
-- Checking if requested hypervisor 'kvm' is supported on this platform ... OK
-- Checking if KVM driver is installed ... 
   Driver is available at /usr/local/bin/docker-machine-driver-kvm ... 
   Checking driver binary is executable ... OK
-- Checking if Libvirt is installed ... OK
-- Checking if Libvirt default network is present ... OK
-- Checking if Libvirt default network is active ... OK
-- Checking the ISO URL ... OK
Error starting the cluster: Error attempting to download and cache 'oc': Cannot get the OpenShift release version v3.7.1: GET https://api.github.com/repos/openshift/origin/releases/tags/v3.7.1: 403 API rate limit exceeded for 66.187.233.202. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.); rate reset in 28m29.895222134s

Also,  you shouldn't have to provide a github token IMHO it is not really automated then and you require folks to pre-setup that part.  We have automated our complete workflow and this takes a step back for us.  
How can we proceed?

**Comment by LalatenduMohanty** [comment]
@arilivigni The rate limit you are seeing is from https://api.github.com/repos/openshift/origin/releases/tags/v3.7.1 i.e. you are trying to download a lot from that location.  If you can get the oc binary in to ~/.minishift/cache/oc/v3.7.1/darwin/oc (for macOS) you might by pass the error as it wont try to download the oc binary.

**Comment by arilivigni** [comment]
@LalatenduMohanty Ok I will try that thanks!

**Comment by anjannath** [comment]
we could just skip the check by setting `warn-check-openshift-release` to true by default. But we'd loose the validation, as in ... if it fails in online mode also, we'll just warn...but the check failed in online mode.

But we do this for some checks... like the Ping check and http check.

**Comment by anjannath** [comment]
resolved via #2149

**Comment by anjannath** [comment]
Opps.. accidentally closed!

**Comment by gbraad** [comment]
perhaps we should error in a specific test, just like happens in the ping test... but this is a PreStart connectivity test from the host... while the current ping is a PostStart connectitivy test from the VM.

@anjannath yes, these test exist, but from a different persepctive. they currently test VM to outside connectivity. this was done to find issues with Hyper-V non-external switched and failures that occured with Xhyve and DNS.

**Comment by anjannath** [comment]
But do we want to bail if the test fails or carry on just by giving a WARN https://github.com/minishift/minishift/issues/2102#issuecomment-374162882 , i guess we should just bail.. as if there's no connectivity or oc is not there in github we cannot proceed, whereas the post checks just WARN ing is fine as we have already come too far.. started the VM already.

**Comment by gbraad** [comment]
we can not bail or break... only WARN.

imagine an offline work situation? We might have cached container images

**Comment by anjannath** [comment]
Yes right, in a situation where `oc` and origin images are cached these will not affect minishift start.

**Comment by LalatenduMohanty** [comment]
Resolved via #2149
